doc/random.adoc

```diff
@@ -0,0 +1,193 @@
+(ideally not at all, since the engine-side seeding should suffice).
+
+If you need `math.random` for its performance but want it to be deterministic,
+you may *reseed* the random after you're done with it to ensure that it is "non-deterministic" again.
+
+[source,lua]
+---
+-- Use the random to generate a seed for the random; preferable over using system time,
+-- as the latter may be deterministic
+local seed = ... -- some fixed seed
+local reseed = math.random(2^31-1)
+math.randomseed(seed) -- temporarily make the random "deterministic"
+-- ... do something using `math.random` ...
+math.randomseed(reseed)
+---
+====
+
+=== https://www.lua.org/manual/5.1/manual.html#pdf-math.random[`math.random`]
+
+Get a random number. Very versatile; allows getting floats between `0` and `1` or integers in a range.
+
+NOTE: The random numbers between `0` and `1` do not provide a full 52-bit mantissa
+full of entropy; they usually have around 32 bits of entropy.
+
+WARNING: When using this to obtain integers,
+make sure that both the upper & lower bound
+as well as their difference are within the C `int` range -
+otherwise you may get overflows & errors.
+
+TIP: Use `math.random` as your go-to versatile "non-deterministic" random source.
+
+== Random Number Generators
+
+=== `PcgRandom`
+
+A seedable 32-bit signed integer pseudo-random number generator.
+
+==== `PcgRandom(seed)`
+
+Constructs a `PcgRandom` instance with the given seed,
+which should be an integer within 32-bit bounds.
+
+==== Methods
+
+===== `:next([min, max])`
+
+If `min` and `max` are both omitted,
+they default to `-2^31` (`-2147483648`)
+and `2^31 - 1` (`2147483647`) respectively.
+
+===== `:rand_normal_dist(min, max, [num_trials])`
+
+WARNING: No successful use of this function is documented. Consider implementing your own normal distribution instead.
+
+`min` and `max` are required; they need to be integers.
+
+Rough approximation of a normal distribution with a mean of `(max - min) / 2`
+and a variance of `(((max - min + 1) ^ 2) - 1) / (12 * num_trials)`.
+
+`num_trials` defaults to `6`. The more trials, the better the approximation.
+
+The return value is a float.
+
+=== `PseudoRandom`
+
+A seedable 16-bit unsigned integer pseudo-random number generator.
+
+"Uses a well-known LCG algorithm introduced by K&R."
+
+Perhaps the lowest-quality random generator of all.
+
+==== `PseudoRandom(seed)`
+
+Constructor: Takes a `seed` and returns a `PseudoRandom` object.
+
+===== `:next([min, max])`
+
+If `min` and `max` are both omitted, they default to `0` and `2^16-1` (`32767`) respectively.
+
+WARNING: Requires `((max - min) == 32767) or ((max-min) <= 6553))` for a proper distribution.
+
+=== `SecureRandom`
+
+System-provided cryptographically secure random:
+An attacker should not be able to predict the generated sequence of random numbers.
+Use this when generating cryptographic keys or tokens.
+
+Not necessarily available in all builds and on all platforms.
```

Comment:
Probably should elaborate on this. On Windows it uses Win32's crypto API, and on every other platform it uses `/dev/urandom` which should exist on everywhere else (including Android).